File: 1137/1137.js
/**
 * @param {number} n
 * @return {number}
 */
var tribonacci = function (n) {
  const mem = [0, 1, 1];
  if (n < 3) return mem[n];
  for (let i = 3; i <= n; i++) {
    mem.push(mem[i - 1] + mem[i - 2] + mem[i - 3]);
  }
  return mem[n];
};

File: 1137/1137.cpp
class Solution {
public:
    int tribonacci(int n) {
        vector<int> mem = {0,1,1};
        if(n<2) return mem[n];
        for(int i=3; i<=n; i++){
            int next = mem[0]+mem[1]+mem[2];
            mem[0] = mem[1];
            mem[1] = mem[2];
            mem[2] = next;
        }
        return mem[2];
    }
};